src/Authoring/WinRT.SourceGenerator/DiagnosticUtils.cs

```diff
@@ -31,15 +31,21 @@ public WinRTComponentScanner(GeneratorExecutionContext context, string assemblyN
         public void FindDiagnostics()
         { 
             HasInvalidNamespace();
-            HasSomePublicTypes();
-            
-            foreach (SyntaxTree tree in _context.Compilation.SyntaxTrees)
+            HasSomePublicTypes();
+
+            WinRTSyntaxReciever syntaxReciever = (WinRTSyntaxReciever)_context.SyntaxReceiver;
+            foreach (var declaration in syntaxReciever.Declarations)
             {
-                var model = _context.Compilation.GetSemanticModel(tree);
-                var nodes = tree.GetRoot().DescendantNodes();
+                var model = _context.Compilation.GetSemanticModel(declaration.SyntaxTree);
 
-                var classes = nodes.OfType<ClassDeclarationSyntax>().Where(IsPublic);
-                foreach (ClassDeclarationSyntax @class in classes)
+                // Check symbol information for whether it is public to properly detect partial types
+                // which can leave out modifier.
+                if(model.GetDeclaredSymbol(declaration).DeclaredAccessibility != Accessibility.Public)

```

Comment:
Does this mean non-public structs will be skipped already? Once you merge I can try those tests I made recently here and check -- might not need my struct PR

Reply:
Also add a space between `if` and `(model.GetDeclaredSymbol`

Reply:
Yes non public structs should be skipped.

Reply:
Also did a format document on this and the other files.